### ISSUE TITLE
Path enhancements

### ISSUE THREAD

**Issue by SCiarella** [GEMDAT-repos/GEMDAT#275]
- [ ] `Pathway` objects should return their total length in Angstrom
- [ ] `Pathway` should also compute and report the energy difference between connected sites, and report this information which is more relevant than the total free energy of the sites
- [ ] The current implementation for `multiple_paths` is inefficient if the user wants paths with large differences. The reason is that the algorithm is based on `netwokx.all_shortest_paths` which starts the search from the optimal one with small variations. We should address that:
- - [ ] the documentation of `gemdat.multiple_paths` should report that it is suited only for small path variations, 
- - [ ] an approach with graph pruning may be optimal to identify strongly distinct paths, but this may deserve a separate issue
- - [ ] the shaded background in the `path_energy` plots representing the sites explored, should be made optional in case of multiple paths that explore different sites

**Comment by stefsmeets** [comment]
I would also like to suggest these changes to `Pathway`:

- [ ] Pathway should have a lattice
- [ ] Pathway should have a dims (`volume.dims()`)

I think this will help decouple `Pathway` from `Volume` so that it can stand better on its own.

**Comment by stefsmeets** [comment]
> * `Pathway` should also compute and report the energy difference between connected sites, and report this information which is more relevant than the total free energy of the sites

@SCiarella Is this essentially `np.diff(path.energy)`? Does this need its own method?

**Comment by SCiarella** [comment]
> > * `Pathway` should also compute and report the energy difference between connected sites, and report this information which is more relevant than the total free energy of the sites
> 
> @SCiarella Is this essentially `np.diff(path.energy)`? Does this need its own method?

Probably a dedicated method is not too useful, but it could be nice to have a plot for the energy difference along the path instead of the total energy.

**Comment by stefsmeets** [comment]
For 3. the shaded background can be turned off by not providing a `structure` since #313

**Comment by SCiarella** [comment]
Once #323 is merged, this issue can be closed.
Task 2. has not been addressed and will be left for a future issue if the need arises.
